heartpoints.sh

```diff
@@ -500,89 +505,166 @@ heartpoints_testUntilSuccess() { local timeoutSeconds=$1; local interval=$2; loc
-heartpoints_minikubeBuildDeployTest() {
+hp_minikubeBuildDeployTest_help() { echo "minikubeBuild, then minikubeDeployTest"; }
+hp_minikubeBuildDeployTest() {
     local shaToBuild="$(git_currentSha)"
-    local taggedImageName="$(heartpoints_minikubeTaggedImageName ${shaToBuild})"
-    heartpoints_minikubeBuild "${taggedImageName}" "${shaToBuild}"
-    heartpoints_minikubeDeployTest "${taggedImageName}"
+    local taggedImageName="$(hp_minikubeTaggedImageName ${shaToBuild})"
+    hp_minikubeBuild "${taggedImageName}" "${shaToBuild}"
+    hp_minikubeDeployTest "${taggedImageName}"
 }
 
-heartpoints_minikubeTaggedImageName() { local shaToBuild=$1
+hp_minikubeTaggedImageName() { local shaToBuild=$1
     requiredParameter "shaToBuild" "${shaToBuild}"
     local imageRepository="minikube"
-    echo "$(heartpoints_taggedImageName ${imageRepository} ${shaToBuild})"
+    echo "$(hp_taggedImageName ${imageRepository} ${shaToBuild})"
 }
 
-heartpoints_minikubeBuild_help() { echo "<taggedImageName> using minikube's docker daemon, build image and tag with minikube metadata"; }
-heartpoints_minikubeBuild() { local taggedImageName=$1; local shaToReportInHttpHeaders=$2
+hp_minikubeBuild_help() { echo "<taggedImageName> using minikube's docker daemon, build image and tag with minikube metadata"; }
+hp_minikubeBuild() { local taggedImageName=$1; local shaToReportInHttpHeaders=$2
     requiredParameter "taggedImageName" "${taggedImageName}"
     requiredParameter "shaToReportInHttpHeaders" "${shaToReportInHttpHeaders}"
-    heartpoints_pointToAndRunMinikubeDockerDaemon
-    heartpoints_pointToAndRunMinikubeDockerDaemon
-    heartpoints_buildAndTagImage "${taggedImageName}" "${shaToReportInHttpHeaders}"
+    hp_pointToAndRunMinikubeDockerDaemon
+    hp_pointToAndRunMinikubeDockerDaemon
+    hp_buildAndTagImage "${taggedImageName}" "${shaToReportInHttpHeaders}"
+}
+
+hp_minikubeDestroyEnvironment_help() { echo "if minikube dev environment is running, destroys it"; }
+hp_minikubeDestroyEnvironment() {
+    hp_minikube delete
+}
+
+hp_urlOfMinikubeWebsite() {
+    echo "https://$(hp_minikube ip)"
+}
+
+hp_minikubeOpenWebsite_help() { echo "assuming site is running in minikube locally, open web browser to home page"; }
+hp_minikubeOpenWebsite() {
+    open "$(hp_urlOfMinikubeWebsite)"
+}
+
+hp_log_path() { local remainingPath=$1
+    local logFile="$(createAndReturnPath "$(devEnvironmentPath)/logs")/${remainingPath}"
+    touch "${logFile}"
+    echo "${logFile}"
+}
+
+brew_package_run() { local packageName=$1; local args="${@:2}"
+    local brewPackagePath="$(brew_package_path ${packageName})"
+    if command_does_not_exist "${brewPackagePath}"; then
+        hp_brew install "${packageName}" > "$(hp_log_path "brewPackageInstall_${packageName}.log")" 2>&1
+    fi
+    "${brewPackagePath}" "${@:2}"
+}
+
+brew_package_path() { local packageName=$1;
+    echo "$(hp_brew --prefix "${packageName}")/bin/${packageName}"
+}
+
+hp_slack() {
```

Comment:
Slack isn't technically a requirement, and this function isn't referenced anywhere (including the README). I think it can be deleted since we want to keep this `.sh` file as small as we can

Reply:
I can add it to the readme, because I am trying to codify and automated everything needed to work with us, including setting up slack. will also add our slack url to the output. i'll add something called "setupDevEnvironment" that can generally include this and other items and then refer to the broad command in the readme.